File: benchmarks/reference/lua/inline_stress.lua
-- inline_stress.lua -- Inlining stress test (500K iterations).
-- Equivalent to examples/il/benchmarks/inline_stress.il

local function double(x)
    return x + x
end

local function square(x)
    return x * x
end

local function add3(a, b, c)
    return a + b + c
end

local function inc(x)
    return x + 1
end

local function combine(x)
    return add3(double(x), square(x), inc(x))
end

local sum = 0
for i = 0, 499999 do
    local r = combine(i)
    local raw_sum = sum + r
    sum = raw_sum & 268435455
end

os.exit(sum & 0xFF)

File: benchmarks/reference/rust/inline_stress.rs
fn double(x: i64) -> i64 {
    x + x
}

fn square(x: i64) -> i64 {
    x * x
}

fn add3(a: i64, b: i64, c: i64) -> i64 {
    a + b + c
}

fn inc(x: i64) -> i64 {
    x + 1
}

fn combine(x: i64) -> i64 {
    add3(double(x), square(x), inc(x))
}

fn main() {
    let mut sum: i64 = 0;
    for i in 0..500000_i64 {
        let r = combine(i);
        let raw_sum = sum + r;
        sum = raw_sum & 268435455;
    }
    std::process::exit((sum & 0xFF) as i32);
}
